tests/test_dsp.py

```diff
@@ -583,6 +583,162 @@ def test_minimum_phase_multidim():
     npt.assert_allclose(imp_minphase.time, imp_zerophase.time, atol=1e-10)
 
 
+def test_impulse_response_delay():
+    """Test delay of an ideal impulse"""
+    n_samples = 2**10
+    snr = 60
+    start_sample = np.array([24])
+
+    ir = pf.signals.impulse(n_samples, delay=start_sample)
+    noise = pf.signals.noise(n_samples, rms=10**(-snr/20), seed=1)
+
+    start_sample_est = dsp.find_impulse_response_delay(ir)
+    npt.assert_allclose(start_sample_est, start_sample, atol=1e-6)
+
+    ir_awgn = ir + noise
+    start_sample_est = dsp.find_impulse_response_delay(ir_awgn)
+    npt.assert_allclose(start_sample_est, start_sample, atol=1e-2)
+
+
+def test_impulse_response_delay_sinc():
+    """Test delay of a band-limited sinc function shifted by 1/2 samples"""
+    sr = 44100
+    n_samples = 128
+    samples = np.arange(n_samples)
+    delay_samples = n_samples // 2 + 1/2
+
+    sinc = np.sinc(samples - delay_samples)
+    win = sgn.get_window('hann', n_samples, fftbins=False)
+
+    ir = pf.Signal(sinc*win, sr)
+    start_samples = pf.dsp.find_impulse_response_delay(ir)
+    npt.assert_allclose(start_samples, delay_samples, atol=1e-3, rtol=1e-4)
+
+
+def test_impulse_response_delay_multidim():
+    """Ideal multi-dimensional Signal of ideal impulses"""
+    n_samples = 2**10
+    snr = 60
+
+    start_sample = [[14, 12, 16], [24, 5, 43]]
+    ir = pf.signals.impulse(n_samples, delay=start_sample)
+
+    noise = pf.signals.noise(n_samples, rms=10**(-snr/20), seed=1)
+
+    ir_awgn = ir + noise
+    start_sample_est = dsp.find_impulse_response_delay(ir_awgn)
+
+    npt.assert_allclose(start_sample_est, start_sample, atol=1e-2)
+
+
+def test_impulse_response_start_insufficient_snr():
+    n_samples = 2**9
```

Comment:
Could be reduced to something like `ir_noise = pf.signals(impulse) + pf.signals(noise, seed=7) * * 10**(-snr/20)`